GVFS/GVFS.Common/Database/SparseTable.cs

```diff
@@ -68,22 +67,27 @@ public HashSet<string> GetAll()
             }
         }
 
-        public void Remove(string directory)
+        public void Remove(string directoryPath)
         {
             try
             {
                 using (IDbConnection connection = this.connectionPool.GetConnection())
                 using (IDbCommand command = connection.CreateCommand())
                 {
                     command.CommandText = "DELETE FROM Sparse WHERE path = @path;";
-                    command.AddParameter("@path", DbType.String, directory);
+                    command.AddParameter("@path", DbType.String, NormalizePath(directoryPath));
                     command.ExecuteNonQuery();
                 }
             }
             catch (Exception ex)
             {
-                throw new GVFSDatabaseException($"{nameof(SparseTable)}.{nameof(this.Remove)}({directory}) Exception: {ex.ToString()}", ex);
+                throw new GVFSDatabaseException($"{nameof(SparseTable)}.{nameof(this.Remove)}({directoryPath}) Exception: {ex.ToString()}", ex);
             }
         }
+
+        private static string NormalizePath(string path)
+        {
+            return path.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar).Trim(Path.DirectorySeparatorChar, ' ');

```

Comment:
Minor: If you call `Trim()` rather than passing in `' '` `Trim()` will trim all characters for which [`Char.IsWhiteSpace`](https://docs.microsoft.com/en-us/dotnet/api/system.char.iswhitespace?view=netframework-4.8) would return `true`.

Reply:
My question was should it be `.Trim().Trim(Path.DirectorySeparatorChar)` or `.Trim(Path.DirectorySeparatorChar).Trim()` or `.Trim().Trim(Path.DirectorySeparatorChar).Trim()` or ...

Reply:
Went with `.Trim().Trim(Path.DirectorySeparatorChar)` since if someone put a space before the directory separator then it will stay but that seems like a user error for the path to me.